main.go

```diff
@@ -132,77 +137,82 @@ func getProjectID(baseURL string, token string, projectname string, namespace st
 }
 
 // Get and return currently active milestones
-func getMilestones(baseURL string, token string, projectID string) (map[string]string, error) {
-	milestones := []milestoneAPI{}
-
-	strURL := []string{baseURL, projectID, "/milestones?state=active&per_page=100"}
-	URL := strings.Join(strURL, "")
+func getActiveMilestones(baseURL string, token string, projectID string) ([]milestoneAPI, error) {
+	state := "active"
+	return getMilestones(baseURL, token, projectID, state)
+}
 
-	m, linkHeader, err := callServer(URL, token, milestones)
-	if err != nil {
-		return m, err
-	}
+// Get and return inactive milestones
+func getInactiveMilestones(baseURL string, token string, projectID string) ([]milestoneAPI, error) {
+	state := "closed"
+	return getMilestones(baseURL, token, projectID, state)
+}
 
-	for _, milestone := range milestones {
-		m[milestone.Title] = milestone.DueDate
-	}
+func getMilestones(baseURL string, token string, projectID string, state string) ([]milestoneAPI, error) {
+	m := []milestoneAPI{}
+	client := &http.Client{}
+	strURL := []string{baseURL, "/projects/", projectID, "/milestones"}
+	URL := strings.Join(strURL, "")
```

Comment:
It would be better to use another option for this process. Because you are making a new slice and it takes more memory. Check the blog post I mentioned in the previous comment for more information.

Reply:
What would be your way to solve this. Always great for people starting out to see the various solutions in code. cc @dbriesz

Reply:
I was thinking about using bytes.Buffer, using bytes.Buffer makes the code longer and a little harder to read, but since you said readability is more important, I think it doesn't need any changes.

Reply:
Yeah you are right, I didn't think about it that way 😄